### DISCUSSION TITLE
Lesson 2: SimpleStorage array question

### DISCUSSION THREAD

**Discussion by ghost** [smartcontractkit/full-blockchain-solidity-course-py#86] [Q&A] [answered]
Hi, is there a difference from doing this as in the lesson:

`return SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).retrieve();`
  
and just this?:

`return simpleStorageArray[_simpleStorageIndex].retrieve();`

I can see the address in the first instance being required when we need to reference a contract that exists elsewhere, but in our case we already have a reference in our array, so can we just use that?

Thanks for any help in understanding!

**Comment by AlbertoSinigaglia** [comment]
Probably you don't need the `address` call, but I'm pretty sure that you need to cast that address to `SimpleStorage` otherwise solidity doesn't know the ABI of that contract, and since solidity is statically typed, it must know that

**Comment by PatrickAlphaC** [comment, marked as answer]
Actually, he's right. 

They are exactly the same. I just wanted to be explicit.

**Reply by ghost** [reply]
Great.  The content is incredible, thank you.
